src/chapters/aspdotnet-model-validation/validation-controller.rst

```diff
@@ -0,0 +1,100 @@
+#. ``NewEvent()`` is called with ``newEvent``
+#. ``newEvent`` is saved
+#. A redirect response is returned, redirecting the user to ``/Events``
+
+The request creates an ``Event`` object using data from the incoming request.
+Regardless of what the data looks like, the new object is saved to the data layer.
+The user could submit an empty form, with no name or description filled in, and our code would be happy to create an ``Event`` and save it.
+Similarly, a user could submit the full text of the massive novel "War and Peace" as the description.
+This isn't great. 
+
+.. admonition:: Note
+
+   Technically, submitting a request containing "War and Peace" would fail with most applications.
+   This is because web servers typically set a limit on the maximum size of a ``POST`` request.
+   However, our application code is willing to take requests of any size, at this point.
+
+Some modest validation rules for a new ``Event`` object might be:
+
+- The ``name`` field must contain between 3 and 20 characters, and 
+- The ``description`` field may be empty, but may contain no more than 500 characters.
+
+With these rules in place, conceptually, the flow of our controller code should look more like the following:
+
+#. Server receives ``POST`` request
+#. Server creates ``newEvent`` object using request parameters
+#. ``NewEvent()`` is called with ``newEvent``
+#. **Controller checks for validation errors in the model object. If errors are found, return the user to the form. Otherwise, proceed.**
+#. ``newEvent`` is saved
+#. A redirect response is returned, redirecting the user to ``/Events``
+
+Let's look at how we can practically do this within ASP.NET Core MVC.
+
+Handling Validation Errors - Video
+----------------------------------
+
+.. TODO: Add video here
+
+.. admonition:: Note
+
+   If you want to verify what code this video starts with, check out the `validation-attributes <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/validation-attributes>`__ branch.
+   If you want to verify what code this video ends with, check out the `handling-errors <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/handling-errors>`__ branch.
+
+Handling Validation Errors - Text
+----------------------------------
+
+When using model binding, we can use tools to validate new model objects before they are saved to a data layer or database. 
+
+Using ``ModelState.IsValid``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Within ``EventController``, the ``NewEvent()`` action method uses model binding to receive an ``Event`` object that is created using form data.
+This object is NOT validated automatically, even if validation attributes are present on its fields.
+
+Recall that *both* the model and controller play a role in validation.
+The model's responsibility is simply to define validation rules.
+The controller must check that those rules are satisfied.
+
+``ModelState.IsValid`` will check in the constraints on the model properties are met.
```

Comment:
I'm in favor of showing the code for this section as well.

Reply:
Done!